raft/leaderElection/leaderelection_test.go

```diff
@@ -0,0 +1,45 @@
+package leaderelection
+
+import (
+	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/types"
+	"testing"
+	"time"
+)
+
+func TestIncrementTermAndBecomeCandidate(t *testing.T) {
+}
+
+func TestGetTimer(t *testing.T) {
+	minTimeout := 150
+	maxTimeout := 650
+
+	// Test if timer is within expected bounds
+	timer := getTimer(minTimeout, maxTimeout)
+	timerDuration := timer.getDuration()
+
+	if timerDuration < minTimeout*Time.Millsecond {
+		t.Error("Timer runs for less time than expected")
+	} else if timerDuration > maxTimeout*Time.Millisecon {
+		t.Error("Timer runs for more time than expected")
+	}
+
+	// Test if timer is random
+}
+
+func TestFindLeader(t *testing.T) {
+	// How do I simulate votes?
+}
+
+func (t *Timer) getDuration() Duration {
+	startTime := time.Now()
+	<-timer.C
+	endTime := time.Now()
+	return endTime.sub(startTime)
+}
+
+func getConfig() types.Configuration {
+	// Assume the config file exists - Would be checked by main_test
```

Comment:
`types.Configuration` in the parameters goes through changes in both main.go and raft.go. I don't want to copy that code here. 

Can we provide helper functions in both packages -
 @NKH123 @apoorva